utils/doclint/xmlDocumentation.js

```diff
@@ -0,0 +1,197 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// @ts-check
+
+const Documentation = require('./documentation');
+const { visitAll } = require('../markdown');
+/**
+ * @param {Documentation.MarkdownNode[]} nodes
+ * @param {number} maxColumns
+ */
+function renderXmlDoc(nodes, maxColumns) {
+    const summary = [];
+    const examples = [];
+    /** @type {Documentation.MarkdownNode} */
+    let lastNode;
+
+    summary.push('<summary>');
+
+    visitAll(nodes, (node, depth) => {
+        lastNode = innerRenderXmlNode(node, lastNode, summary, examples, maxColumns);
+    });
+
+    if (summary.length == 1) { // just the <summary> node 
+        return [];
+    }
+
+    // we might have a stray list, if the <li> is the last element
+    if (lastNode && lastNode.type === 'li') {
+        summary.push('</list>');
+    }
+    summary.push('</summary>');
+
+    // add examples
+    summary.push(...examples);
+    return summary.map(n => `/// ${n}`);
+}
+
+/**
+ * @param {Documentation.MarkdownNode} node
+ * @param {Documentation.MarkdownNode} lastNode
+ * @param {number=} maxColumns
+ * @param {string[]} summary
+ * @param {string[]} examples
+ */
+function innerRenderXmlNode(node, lastNode, summary, examples, maxColumns) {
+    /** @param {string[]} a */
+    const newLine = (a) => {
+        if (a[a.length - 1] !== '')
+            a.push('');
+    };
+
+    let escapedText = node.text;
+    // resolve links (within [])
+
+    if (node.type === 'text') {
+        // clear up the list, if there was one
+        if (lastNode && lastNode.type === 'li') {
```

Comment:
nit: we don't wrap single-liners with {}

Reply:
I've mostly avoided doing that now, but then `eslint` seems to have thrown a fit. Either way, point taken, thanks!